File: src/main/java/org/pixeltime/enchantmentsenhance/command/ReloadCommand.java
package org.pixeltime.enchantmentsenhance.command;

import org.bukkit.entity.Player;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class ReloadCommand extends SubCommand {

    @Override
    public void onCommand(Player player, String[] args) {
        new ReloadCommand().onCommand(player, args);
    }


    @Override
    public String name() {
        return "reload";
    }


    @Override
    public String info() {
        return "\n&6/enhance reload &7- " + SettingsManager.lang.getString(
                "Help.reload");
    }


    @Override
    public String[] aliases() {
        return new String[]{"reload", "rel", "chongzai", "cz"};
    }


    @Override
    public String getPermission() {
        return "Enchantmentsenhance.reload";
    }

}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/command/player/ReloadCommand.kt
/*
 *     Copyright (C) 2017-Present HealPotion
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 */

package org.pixeltime.enchantmentsenhance.command.player

import org.bukkit.entity.Player
import org.pixeltime.enchantmentsenhance.Main
import org.pixeltime.enchantmentsenhance.command.SubCommand
import org.pixeltime.enchantmentsenhance.manager.SettingsManager
import org.pixeltime.enchantmentsenhance.util.Util

class ReloadCommand : SubCommand() {


    override val permission: String
        get() = "Enchantmentsenhance.reload"

    override fun onCommand(player: Player, args: Array<String>) {
        SettingsManager.reloadConfig()
        SettingsManager.reloadData()
        SettingsManager.reloadLang()
        Main.getMain().registerDataSettings()
        Util.sendMessage(SettingsManager.lang.getString("Config.reload"),
                player)
    }


    override fun name(): String {
        return "reload"
    }


    override fun usage(): String {
        return "/enhance reload"
    }


    override fun aliases(): Array<String> {
        return arrayOf("reload", "rel", "chongzai", "cz")
    }

}
